landoapi/workers/landing_worker.py

```diff
@@ -86,11 +86,37 @@ def __init__(self, *args, **kwargs):
         self.last_job_finished = None
         self.refresh_enabled_repos()
 
-    def loop(self):
-        logger.debug(
-            f"{len(self.applicable_repos)} applicable repos: {self.applicable_repos}"
+    def check_landing_worker_warnings(self):
+        """Log messages that show various important statistics about the landing worker."""
+        TOO_MANY_ATTEMPTS_THRESHOLD = 10
+        # TODO: should this threshold be different for try landing worker?
+        QUEUE_SIZE_THRESHOLD = 20
+        ACTIVE_STATUSES = (
+            LandingJobStatus.SUBMITTED,
+            LandingJobStatus.DEFERRED,
+            LandingJobStatus.IN_PROGRESS,
+        )
+
+        queue_size = LandingJob.query.filter(
+            LandingJob.status.in_(ACTIVE_STATUSES)
+        ).count()
+        if queue_size >= QUEUE_SIZE_THRESHOLD:
+            logger.warning(f"The landing queue size of {queue_size} exceeds threshold")
+
+        runaway_jobs = LandingJob.query.filter(
+            LandingJob.status.in_(ACTIVE_STATUSES),
+            LandingJob.attempts >= TOO_MANY_ATTEMPTS_THRESHOLD,
         )
 
+        if runaway_jobs.count() > 0:
+            job = runaway_jobs.all()[0]
+            logger.warning(
+                f"Active landing job ({job}) has too many attempts ({job.attempts})"
+            )
```

Comment:
Should we cancel the job in that case?

Reply:
That would be a good idea, but I think it would be outside the scope of this PR since it would require us to also send a notification email. Maybe it can be done as a follow up.